crypto/ube/ube_test.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include  <gtest/gtest.h>
+
+#include "internal.h"
+
+TEST(Ube, BasicTests) {
+  uint64_t generation_number = 0;
+  ASSERT_TRUE(get_ube_generation_number(&generation_number));
+
+  // Check stability.
+  uint64_t current_generation_number = generation_number + 1;
+  ASSERT_TRUE(get_ube_generation_number(&current_generation_number));
+  ASSERT_EQ(current_generation_number, generation_number);
+
+  // Check stability again.
+  current_generation_number = generation_number + 2;
+  ASSERT_TRUE(get_ube_generation_number(&current_generation_number));
+  ASSERT_EQ(current_generation_number, generation_number);
+}
+
+TEST(Ube, MockedMethodTests) {
+  uint64_t generation_number = 0;
+  uint64_t cached_generation_number = 0;
+  if (get_ube_generation_number(&generation_number) == 0) {
+    // In this case, UBE detection is disabled, so just return
+    // successfully. The short-circuit feature means we can't mock detection
+    // methods.
+    return;
+  }
+
+  // The fork generation number is initially 1. Add a larger offset to avoid
+  // any noise. 
```

Comment:
What noise is there in this nest?

Reply:
This is a left over thought... Nothing right now! But tests are run concurrently, and stuff under `ube_get_detection_generation_numbers` is global. So, we need to be a bit careful when writing these tests because direct modification of the global state will affect other tests that might expect different behaviour.

Reply:
I changed the wording. But I planned to tackle this in subsequent work.